src/server/plugins/engine/models/SummaryViewModel.ts

```diff
@@ -190,6 +166,51 @@ export class SummaryViewModel {
   }
 }
 
+function addRepeaterItem(
+  page: RepeatPageController,
+  request: Request,
+  model: FormModel,
+  state: FormSubmissionState,
+  items: DetailItem[]
+) {
+  const { options } = page.repeat
+  const { name, title } = options
+  const repeatSummaryPath = page.getSummaryPath(request)
+  const path = `/${model.basePath}${page.path}`
+  const rawValue = state[options.name]
+  const isInitialised = Array.isArray(rawValue)
+  const value = isInitialised ? rawValue.length.toString() : '0'
+  const url = redirectUrl(request, isInitialised ? repeatSummaryPath : path, {
+    returnUrl: redirectUrl(request, `/${model.basePath}/summary`)
+  })
+
+  const subItems: DetailItem[][] = []
+
+  if (isInitialised) {
+    rawValue.forEach((itemState) => {
+      const sub: DetailItem[] = []
+      for (const component of page.components.formItems) {
+        const item = Item(request, component, itemState, page, model)
```

Comment:
With us drilling down from `state` into `itemState`

We've still got `Item()` typed to accept `FormSubmissionState`

Do we need safeguards to make sure this never blows up?

```js
rawValue: state[component.name],
```

Reply:
I'd argue that would be exceptional circumstances. I'm happy to do something like:

`const item = Item(request, component, (itemState ?? {}), page, model)`

but I just can't see how that could happen and in the event of an undefined/null item getting into the repeater array, it would probably warrant an exception.

Reply:
Only flagging it because it's no longer `FormSubmissionState`

E.g. It definitely doesn't have `progress: []` in there

1. Payload `FormPayload` (including `crumb`) extends `FormData`
2. Redis state `FormSubmissionState` extends `FormState` perhaps

We're not necessarily fixing issues here, we're putting the safeguards in place for future work

Can leave it for now